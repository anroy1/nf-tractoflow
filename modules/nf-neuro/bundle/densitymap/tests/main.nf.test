nextflow_process {

    name "Test Process BUNDLE_DENSITYMAP"
    script "../main.nf"
    process "BUNDLE_DENSITYMAP"

    tag "modules"
    tag "modules_nfcore"
    tag "bundle"
    tag "bundle/densitymap"

    tag "subworkflows"
    tag "subworkflows/load_test_data"

    setup {
        run("LOAD_TEST_DATA", alias: "LOAD_DATA") {
            script "../../../../../subworkflows/nf-neuro/load_test_data/main.nf"
            process {
                """
                input[0] = Channel.from( [ "bundles.zip" ] )
                input[1] = "test.load-test-data"
                """
            }
        }
    }

    test("densitymap - map") {

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("densitymap - mask") {

        config "./nextflow_mask.config"

        when {
            process {
                """
                input[0] = LOAD_DATA.out.test_data_directory.map{
                    test_data_directory -> [
                        [ id:'test', single_end:false ], // meta map
                        file("\${test_data_directory}/bundle_all_1mm.trk")
                    ]
                }
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
