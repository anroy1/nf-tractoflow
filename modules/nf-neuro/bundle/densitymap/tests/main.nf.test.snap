{
    "densitymap - map": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd0fdb72386eb4cf4066566dfad0a893"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,a9cc8ba016a680ec8cbac940c45e5464"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd0fdb72386eb4cf4066566dfad0a893"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T14:06:16.074091804"
    },
    "densitymap - mask": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bd0fdb72386eb4cf4066566dfad0a893"
                ],
                "density_map": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__bundle_all_1mm.nii.gz:md5,5c682d295c593e132295e21e753a6f4f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bd0fdb72386eb4cf4066566dfad0a893"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T14:07:53.092923761"
    }
}